wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/MoshiJsonIntegration.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import com.squareup.wire.internal.FieldBinding.JsonFormatter
+import com.squareup.wire.internal.JsonIntegration
+import java.lang.reflect.Type
+
+internal object MoshiJsonIntegration : JsonIntegration<Moshi, JsonAdapter<Any?>>() {
+  override fun frameworkAdapter(
+    framework: Moshi,
+    type: Type
+  ): JsonAdapter<Any?> =
+    framework.adapter<Any?>(type)
+        .nullSafe()
```

Comment:
This doesn't read nice, does it?

```suggestion
  ): JsonAdapter<Any?> {
    return framework.adapter<Any?>(type).nullSafe()
  }
```

Reply:
Fixed.